GVFS/GVFS/RepairJobs/GitHeadRepairJob.cs

```diff
@@ -182,5 +103,17 @@ private bool CanBeRepaired(List<string> messages)
 
             return true;
         }
+
+        private string GetHeadRefSha()
+        {
+            string headRefFilePath = Path.Combine(this.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Head);
+
+            var contents = File.ReadAllText(headRefFilePath);
+            var sha = contents.Trim();
+
+            Debug.Assert(SHA1Util.IsValidShaFormat(sha), "Fix to HEAD ref should have written a valid SHA");

```

Comment:
Also this condition seems invalid. The HEAD file does not have to contain a sha, it can contain the name of another ref.

I realize that you're currently calling this method after writing a sha, but only the calling code knows that, this method does not.